### PR TITLE
Add support for searching other datasources

### PR DESCRIPTION
**Pull request by Marlinc**
This change adds a trait which can be used in custom repository
classes and adds support for custom repositories to the value and
boolean filters.

Signed-off-by: Marlin Cremers <m.cremers@cvo-technologies.com>

### CODE REVIEW

**Comment by ADmad** [comment]
Failing test seems related to this change.

**Comment by ADmad** [comment]
@Marlinc ping

**Comment by Marlinc** [comment]
Working now @ADmad

File: src/Model/SearchTrait.php [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+use Exception;
+use Search\Manager;
+
+trait SearchTrait
+{
+
+    /**
+     * Search Manager instance.
+     *
+     * @var \Search\Manager
```
Comment (dereuromark) [inline src/Model/SearchTrait.php]: Does this need |null ?

Comment (Marlinc) [reply src/Model/SearchTrait.php]: Add it

File: src/Model/Filter/Value.php [hunk 2]
```diff
@@ -27,6 +28,16 @@ public function process()
             return false;
         }
 
+        if (!$this->manager()->getRepository() instanceof Table) {
```
Comment (ADmad) [inline src/Model/Filter/Value.php]: Shouldn't this block go after the multi values check below?

**Comment by ADmad** [comment]
Would be nice if some info was added to the readme about these changes.

**Comment by Marlinc** [comment]
I've added a small text to the README, I hope it will be clear to users

File: src/Model/Filter/Value.php (hunk 2 above)
Comment (Marlinc) [reply src/Model/Filter/Value.php]: Let me check

Comment (Marlinc) [reply src/Model/Filter/Value.php]: Changed :smile:

File: README.md [hunk 3]
```diff
@@ -182,6 +182,35 @@ public function initialize()
 The `Search.Prg` component will allow your filtering forms to be populated using
 the data in the query params. It uses the [Post, redirect, get pattern](https://en.wikipedia.org/wiki/Post/Redirect/Get).
 
+### Custom repository
+
+It is also possible to use the search plugin on custom repositories like
```
[the remark this answers is not in the document]
Comment (Marlinc) [reply README.md]: Done

**Comment by bravo-kernel** [comment]
Aha, finally understand what "other datasources" means, thanks !

> custom repositories which implement `Cake\Datasource\RepositoryInterface`

**Pull request merged**